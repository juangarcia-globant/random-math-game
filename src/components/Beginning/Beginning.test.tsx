import { render, screen, waitFor } from "@testing-library/react"
import { Beginning } from ".";

describe("<Beginning />", () => {
  test("Matches the snapshot", () => {
    const { asFragment } = render(<Beginning time={3} setBeginningDone={() => {}} />);
    expect(asFragment()).toMatchSnapshot()
  });

  test("The beginning component should be render", async () => {
    render(<Beginning time={3} setBeginningDone={() => {}} />);
    const getReadyMsg = screen.getByRole('heading', { level:3 });
    expect(getReadyMsg).toBeInTheDocument();
  });

  test("The numbers should be counting down", async () => {
    render(<Beginning time={3} setBeginningDone={() => {}} />);
    const countingDown = screen.getByRole('heading', { level: 1 });
    await waitFor(() => {
      expect(Number(countingDown.innerHTML)).toBe(2);
    }, {timeout: 1000});

    await waitFor(() => {
      expect(Number(countingDown.innerHTML)).toBe(1);
    }, {timeout: 2000});
  });

  test("mockSetBeginning function should be execute after 3 seconds", async () => {
    const mockSetBeginning = jest.fn(x => {});
    render(<Beginning time={3} setBeginningDone={mockSetBeginning} />);
    await waitFor(() => {
      expect(mockSetBeginning).toHaveBeenCalledTimes(1);
    }, {timeout: 4000})
  });
});