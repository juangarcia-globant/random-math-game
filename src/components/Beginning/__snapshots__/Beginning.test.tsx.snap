// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Beginning /> Matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
       The game will be start in... 
    </h3>
    <h1>
       3 
    </h1>
  </div>
</DocumentFragment>
`;
